packages/gestalt/src/IconButton.js

```diff
@@ -87,9 +107,75 @@ const IconButtonWithForwardRef: React$AbstractComponent<
   const classes = classnames(styles.button, touchableStyles.tapTransition, {
     [styles.disabled]: disabled,
     [styles.enabled]: !disabled,
-    [touchableStyles.tapCompress]: !disabled && isTapping,
+    [touchableStyles.tapCompress]:
+      props.role !== 'link' && !disabled && isTapping,
   });
 
+  function handleClick(event) {
+    if (onClick) {
+      onClick({ event });
+    }
+  }
+
+  function handleLinkClick({ event }) {
+    handleClick(event);
+  }
+
+  const renderPogComponent = (selected?: boolean): Node => {
+    return (
+      <Pog
+        active={!disabled && isActive}
+        bgColor={bgColor}
+        dangerouslySetSvgPath={dangerouslySetSvgPath}
+        focused={!disabled && isFocusVisible && isFocused}
+        hovered={!disabled && isHovered}
+        icon={icon}
+        iconColor={iconColor}
+        padding={padding}
+        selected={selected}
+        size={size}
+      />
+    );
+  };
+
+  if (props.role === 'link') {
+    const { href, rel, target } = props;
+
+    return (
+      <Link
+        accessibilityLabel={accessibilityLabel}
+        disabled={disabled}
+        inline
+        href={href}
+        onClick={handleLinkClick}
+        ref={innerRef}
+        rel={rel}
+        rounding="circle"
+        tapStyle={disabled ? undefined : 'compress'}
+        target={target}
+      >
+        <div
+          onMouseEnter={() => setHovered(true)}
+          onMouseLeave={() => {
+            setActive(false);
+            setHovered(false);
+          }}
+          className={classes}
```

Comment:
Why did we not add `onMouseUp` and `onMouseDown`? It might be good to extra all of these functions (since they are now re-used`.

`onMouseEnter={handleMouseEnter}`

Reply:
DONE